airbyte-integrations/connectors/source-serpstat/source_serpstat/manifest.yaml

```diff
@@ -0,0 +1,584 @@
+        request_body_json:
+          id: '{{ now_utc() }}'
+          method: SerpstatDomainProcedure.getDomainsHistory
+          params:
+            se: '{{config[''region_id'']}}'
+            page: '{{(next_page_token[''next_page_token''] or 0) + 1}}'
+            size: '{{config[''page_size'']}}'
+            sort:
+              date: desc
+            domain: '{{config[''domain'']}}'
+            during_all_time: true
+      record_selector:
+        type: RecordSelector
+        extractor:
+          type: DpathExtractor
+          field_path:
+            - result
+            - data
+      paginator:
+        type: DefaultPaginator
+        page_token_option:
+          type: RequestOption
+          inject_into: header
+          field_name: X-page
+        pagination_strategy:
+          type: CursorPagination
+          cursor_value: '{{response.result.summary_info.page}}'
+          stop_condition: '{{response.result.summary_info.page > config[''pages_to_fetch''] - 1}}'
+  - type: DeclarativeStream
+    name: Domains summary
+    primary_key:
+      - domain
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        $schema: http://json-schema.org/schema#
+        properties:
+          ad_keywords:
+            type: number
+          ads:
+            type: number
+          ads_dynamic:
+            type: number
+          domain:
+            type: string
+          down_keywords:
+            type: number
+          keywords:
+            type: number
+          keywords_dynamic:
+            type: number
+          new_keywords:
+            type: number
+          out_keywords:
+            type: number
+          prev_date:
+            type: string
+          rised_keywords:
+            type: number
+          traff:
+            type: number
+          traff_dynamic:
+            type: number
+          visible:
+            type: number
+          visible_dynamic:
+            type: number
+        type: object
+    retriever:
+      type: SimpleRetriever
+      requester:
+        type: HttpRequester
+        url_base: https://api.serpstat.com/v4/
+        path: /
+        http_method: POST
+        request_parameters: {}
+        request_headers:
+          X-request-sender: Airbyte
+        authenticator:
+          type: ApiKeyAuthenticator
+          api_token: '{{ config[''api_key''] }}'
+          inject_into:
+            type: RequestOption
+            field_name: token
+            inject_into: header
+        error_handler:
+          type: CompositeErrorHandler
+          error_handlers:
+            - type: DefaultErrorHandler
+              backoff_strategies:
+                - type: ExponentialBackoffStrategy
+                  factor: 2
+        request_body_json:
+          id: '{{ now_utc() }}'
+          method: SerpstatDomainProcedure.getDomainsInfo
+          params:
+            se: '{{config[''region_id'']}}'
+            domains: '{{config[''domains'']}}'
```

Comment:
@alex-danilin the connector looks great. I think the only missing part is this endpoint which uses the `domains` instead of `domain`. It is confusing because `domains` is not a required parameter and if I don't send the info the sync will break. I think it works for the parameter `region_id` too because of `Domain Region` endpoint.

Also can you implement pagination to all stream allow it?

Reply:
@marcosmarxm Our API returns the data based on different types of entities. Some API methods require a single domain as an input, some require an array of domains, and others that we plan to add to this connector in the future require a keyword. If I set all these fields as required we will force the users who don't use all the methods to fill these fields in with some arbitrary values, which might be confusing. Can I describe all the required fields for each stream in the connector docs without setting them as required in connector settings?

Reply:
I agree, we can update the documentation! But the pagination working is fundamental.